File: drivingSchool/DAL/Account.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Data;
using System.Data.SqlClient;
using System.Configuration;

using drivingSchool.Model;

namespace drivingSchool.DAL
{
    public class Account
    {
        static public bool isExist(string table,string type,AccountModel user)
        {
            string command = null;
            if (type == "account")
            {
                command = $"select account from {table} where account = '{user.account}'";
            }else if (type == "password")
            {
                command = $"select password from {table} where password = '{user.password}'";
            }else if (type == "user")
            {
                command = $"select account from {table} where password = '{user.password}' and account = '{user.account}'";
            }else if(type == "name"){
                command = $"select name from {table} where name = '{user.name}'";
            }
            try
            {
                if (SQL.getReader(command).HasRows)
                {
                    SQL.Dispose();
                    return true;
                }
                else
                {
                    SQL.Dispose();
                    return false;
                }

            }
            catch
            {
                return false;
            }
           
        }

        static public string getQuestionAnswer(string table,string account)
        {
            string result = "";
            string command = $"select * from {table} where account = '{account}'";
            SqlDataReader read = SQL.getData(command);
            result = read["question"]+"="+read["answer"];
            SQL.Dispose();
            return result;
        }
        static public bool changePwd(string table,string account,string newPassword)
        {
            string command = $"update {table} set password = '{newPassword}' where account = '{account}'";
            return SQL.Excute(command);
        }
        static public bool changeInfo(string table,AccountModel user)
        {
            string command = $"update {table} set name = '{user.name}',address = '{user.address}',description = '{user.description}' where account = '{user.account}'";
            return SQL.Excute(command);
        }
    }
}

File: drivingSchool/BLL/js/component/ajax/account.js
import {accountAjax} from "js/common/variable";
export class Account{
    static hasInfo(type, table,name = "",account = "",password = "") {
        let returnValue = false;
        $.ajax({
            url: accountAjax,
            type: "POST",
            async: false,
            data: {
                "key": "hasInfo",
                "table": table,
                "type": type,
                "name":name,
                "account":account,
                "password":password
            },
            success: result => {
                returnValue = result.toLowerCase() === "true";
            }
        });
        return returnValue;
    }
    
    static changeInfo(table, user) {
        let returnValue = false;
        $.ajax({
            url: accountAjax,
            type: "POST",
            async: false,
            data: {
                "key": "changeInfo",
                "table": table,
                "account": user.act,
                "address": user.address,
                "description": user.description
            },
            success: result => {
                returnValue = result.toLowerCase() === "true";
            }
        });
        return returnValue;
    }
    
    static changePwd(table, account,password) {
        let returnValue = false;
        $.ajax({
            url: accountAjax,
            type: "POST",
            async: false,
            data: {
                "key": "changePwd",
                "table": table,
                "account": account,
                "password": password
            },
            success: result => {
                returnValue = result.toLowerCase() === "true";
            }
        });
        return returnValue;
    }
    
    static getQuestionAnswer(table,account){
        let returnValue = false;
        $.ajax({
            url: accountAjax,
            type: "POST",
            async: false,
            data: {
                "key":"getQuestionAnswer",
                "table": table,
                "account": account
            },
            success: result => returnValue = result.split("=")
        });
        return returnValue;
    }
    
    static register(user){
        let returnValue = false;
        $.ajax({
            url: accountAjax,
            type: "POST",
            async: false,
            data: {
                "key": "register",
                "table": "student",
                "account": user.act,
                "password": user.pwd,
                "question": user.question,
                "answer": user.answer,
                "name": user.name,
                "address": user.address,
                "description": user.description
            },
            success: result => {
                returnValue = result.toLowerCase() === "true";
            }
        });
        return returnValue;
    }
}
